src/lib_db.ml

```diff
@@ -326,3 +337,8 @@ let anonymous_scope t =
     lib_db = t
   ; scope = internal_name_scope t ~dir:t.anonymous_root
   }
+
+let find_scope_by_name_exn t ~name =
+  match Hashtbl.find t.by_scope_name name with
+  | None -> raise (Code_error (sprintf "Invalid scope '%s'" name))
```

Comment:
`Code_error` is for errors that the user shouldn't see. If someone asks for `_build/default/.ppx/ppx_foo@blah/ppx.exe`, they might see this error, so it should be `die "Invalid scope ..."` instead

Reply:
OK, I haven't considered that you could request a target explicitly. Pushed a commit that uses `die`.